src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,162 @@
+ */
+
+using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	// you can provide a topView argument or it will try to find the ContainerViewController
+	internal static void GoToNextResponderOrResign(UIView view, UIView? topView = null)
+	{
+		if (!view.CheckIfEligible())
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var targetSuperview = topView ?? view.GetViewController()?.GetContainerViewController();
+		if (targetSuperview is null)
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var textFields = GetDeepResponderViews(targetSuperview);
+
+		view.MoveToNextField(textFields);
+	}
+
+	static bool CheckIfEligible(this UIView view)
+	{
+		if (view is UITextField field && field.ReturnKeyType == UIReturnKeyType.Next)
+			return true;
+		else if (view is UITextView)
+			return true;
+
+		return false;
+	}
+
+	static UIView? GetContainerViewController(this UIViewController controller)
+	{
+		var curView = controller.View;
+
+		while (curView?.Superview is not null)
+		{
+			var curController = curView.GetViewController();
+			if (curController is ContainerViewController container)
+				return container.View;
+
+			curView = curView.Superview;
+		}
+
+		return null;
+	}
+
+	static void MoveToNextField (this UIView view, List<UIView> textFields)
+	{
+		// get the index of the current textField and go to the next one
+		var currentIndex = textFields.FindIndex(v => v == view);
+		var nextIndex = currentIndex < textFields.Count - 1 ? currentIndex + 1 : -1;
+
+		if (nextIndex != -1)
+			textFields[nextIndex].BecomeFirstResponder();
+		else
+			view.ResignFirstResponder();
+	}
+
+	// Find all of the eligible UITextFields and UITextViews inside this view
+	internal static List<UIView> GetDeepResponderViews(UIView view)
+	{
+		var textItems = view.GetResponderViews();
+
+		textItems.Sort(new ResponderSorter());
+
+		return textItems;
+	}
+
+	static List<UIView> GetResponderViews(this UIView view)
+	{
+		var textItems = new List<UIView>();
+
+		foreach (var child in view.Subviews)
+		{
+			if (child is UITextField textField && child.CanBecomeFirstResponder())
+				textItems.Add(textField);
+
+			else if (child is UITextView textView && child.CanBecomeFirstResponder())
+				textItems.Add(textView);
+
+			else if (child.Subviews.Length > 0 && !child.Hidden && child.Alpha > 0f)
+				textItems.AddRange(child.GetResponderViews());
+		}
+
+		return textItems;
+	}
+
+	static bool CanBecomeFirstResponder(this UIView view)
+	{
+		var isFirstResponder = false;
+
+		if (view is UITextView tview)
+			isFirstResponder = tview.Editable;
+		else if (view is UITextField field)
+			isFirstResponder = field.Enabled;
+
+		return !isFirstResponder ? false :
+			!view.Hidden
+			&& view.Alpha != 0f
+			&& !view.IsAlertViewTextField();
+	}
+
+	static bool IsAlertViewTextField(this UIView view)
```

Comment:
Do we have a scenario where this check is currently relevant? 

I realize it's in the original Swift but if we don't have a use case where this check is being flexed then I think we leave it out? 

Maybe we can just comment out the check on line 115 with a comment to the original code?

Reply:
Yeah I didn't really see this being that useful in our code so I can def comment it out!